### ISSUE TITLE
install rotary fail:no such file or directory

### ISSUE THREAD

**Issue by johnny37** [Dao-AILab/flash-attention#705]
when i use "python setup install", i cause these error, who can help me?

these are logs:
>>>>>>>>
torch.__version__  = 2.1.0+cu121

running install
running bdist_egg
running egg_info
writing rotary_emb.egg-info/PKG-INFO
writing dependency_links to rotary_emb.egg-info/dependency_links.txt
writing top-level names to rotary_emb.egg-info/top_level.txt
reading manifest file 'rotary_emb.egg-info/SOURCES.txt'
writing manifest file 'rotary_emb.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
running build_ext
/home/xm/work/Langchain-Chatchat-master/venv/lib/python3.10/site-packages/torch/utils/cpp_extension.py:424: UserWarning: There are no x86_64-linux-gnu-g++ version bounds defined for CUDA version 12.1
  warnings.warn(f'There are no {compiler_name} version bounds defined for CUDA version {cuda_str_version}')
building 'rotary_emb' extension
Emitting ninja build file /home/xm/work/Langchain-Chatchat-master/flash-attention-2.1.1/csrc/rotary/build/temp.linux-x86_64-3.10/build.ninja...
Compiling objects...
Allowing ninja to set a default number of workers... (overridable by setting the environment variable MAX_JOBS=N)
1.11.1.git.kitware.jobserver-1
x86_64-linux-gnu-g++ -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -g -fwrapv -O2 -Wl,-Bsymbolic-functions -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 /home/xm/work/Langchain-Chatchat-master/flash-attention-2.1.1/csrc/rotary/build/temp.linux-x86_64-3.10/rotary.o /home/xm/work/Langchain-Chatchat-master/flash-attention-2.1.1/csrc/rotary/build/temp.linux-x86_64-3.10/rotary_cuda.o -L/home/xm/work/Langchain-Chatchat-master/venv/lib/python3.10/site-packages/torch/lib -L/usr/local/cuda-12.1/lib64 -lc10 -ltorch -ltorch_cpu -ltorch_python -lcudart -lc10_cuda -ltorch_cuda -o build/lib.linux-x86_64-3.10/rotary_emb.cpython-310-x86_64-linux-gnu.so
/usr/bin/ld: can not find /home/xm/work/Langchain-Chatchat-master/flash-attention-2.1.1/csrc/rotary/build/temp.linux-x86_64-3.10/rotary.o: no such file or directory
/usr/bin/ld: can not find /home/xm/work/Langchain-Chatchat-master/flash-attention-2.1.1/csrc/rotary/build/temp.linux-x86_64-3.10/rotary_cuda.o: no such file or directory
collect2: error: ld returned 1 exit status
error: command '/usr/bin/x86_64-linux-gnu-g++' failed with exit code 1

**Comment by ruio248** [comment]
I also encountered the same problem，have you fixed it yet?

**Comment by johnny37** [comment]
> I also encountered the same problem，have you fixed it yet?

no, i dont know how to do, if you fix it, pleace call me soon, tks.
